code/__HELPERS/unsorted.dm

```diff
@@ -339,6 +339,17 @@ Turf and target are separate in case you want to teleport some distance from a t
 			break
 	return loc
 
+//Returns a list of all locations (except the area) the movable is within.
+/proc/get_nested_locs(atom/movable/AM, include_turf = FALSE)
+	. = list()
+	var/atom/A = AM.loc
+	var/turf/T = get_turf(AM)
+	while(A && A != T)
+		. += A
+		A = A.loc
+	if(A && include_turf) //At this point, only the turf is left, if there is one.
+		. += A
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names